src/main/java/DelimiterParser.java

```diff
@@ -0,0 +1,43 @@
+import java.util.Arrays;
+import java.util.List;
+
+public class DelimiterParser {
+
+    public static final String REGEX_NOT_NUMBER = "\\D+";
+    public static final String REGEX_CAN_CUSTOM_DELIMITER = "^//.+\n.*";
+
+    private List<String> delimiter = Arrays.asList(":", ",");
+
+    public String[] parse(final String expression) {
+        final String updatedExpression = updateDelimiter(expression);
+
+        String[] operands = updatedExpression.split(String.join("|", delimiter));
+
+        checkValidate(operands);
+
+        return operands;
+    }
+
+    private String updateDelimiter(final String expression) {
+        if (canChangeNewDelimiter(expression)) {
+            delimiter = List.of(expression.substring(2, expression.indexOf("\n")));
+            return expression.substring(expression.indexOf("\n") + 1);
+        } else {
+            return expression; // 원래 if 문에 return으로 끝내고 else를 주는 편인데, 이 편이 좀 더 가독성이 좋아보여 else로 해봄
+        }
```

Comment:
이부분에서 else를 이용하는 것보다는 바로 return을 해주는 것이 좋을 것 같습니다!

```suggestion
        if (canChangeNewDelimiter(expression)) {
            delimiter = List.of(expression.substring(2, expression.indexOf("\n")));
            return expression.substring(expression.indexOf("\n") + 1);
        } 
        return expression;
```

Reply:
넵 주석에 달았듯이 저도 if에서 return 시켜서 else를 안 쓰는 방향으로 해보는데요. 이번엔 일부러 한 번 else를 달아봤습니다 ㅎㅎ
그럼에도 역시 else가 없는게 더 좋은거 같네요!
수정해보겠습니다!